shakenfist/net.py

```diff
@@ -99,7 +104,33 @@ def subst_dict(self):
     def persist_floating_gateway(self):
         db.persist_floating_gateway(self.uuid, self.floating_gateway)
 
+    def is_okay(self):
+        """Check if network is created and running"""
+        # TODO(andy):This will be built upon with further code re-design
+
+        okay = self.is_created()
+
+        if is_network_node():
+            okay &= self.is_dnsmasq_running()
+
+        return okay
+
+    def is_created(self):
+        """Attempt to ensure network has been created successfully"""
+
+        subst = self.subst_dict()
+        out, _ = processutils.execute('ip link show %(vx_bridge)s' % subst,
+                                      shell=True)
+        net_created = re.match(r'.*[<,]UP[,>].*', out)
+        if not net_created:
+            LOG.warning('network(%s) is_created() bridge is not up: %s',
```

Comment:
network has a __str__ which should give a nicely formatted description of the network (uuid and vxid). I think that's what we use for the rest of this file? I do think logging needs a rethink though.

Reply:
Sorted.